docs/application/native/guides/app-management/ui-app/dali-widget-app.md

```diff
@@ -0,0 +1,268 @@
+
+<a name="create"></a>
+## The steps to create a DALi widget application:
+
+1. Initialize the widget application:
+
+   Create the Dali::WidgetApplication class instance to initialize and set up DALi for a widget application:
+   ```
+   int DALI_EXPORT_API main( int argc, char **argv )
+   {
+     WidgetApplication application = WidgetApplication::New( &argc, &argv, "" );
+
+     WidgetApplicationExample test( application );
+
+     application.MainLoop();
+
+     return 0;
+   }
+   ```
+
+2. Connect signals to keep yourself informed when certain system events occur.
+
+   To manage signal connection safely, DALi provides the `Dali::ConnectionTracker` class. A typical way to start a DALi application is to create a class derived from the `Dali::ConnectionTracker` class and use its member functions as callback functions for DALi signals. When the application receives the `InitSignal()` function, it can create user interface components. Additionally, connect the callback `WidgetApplicationExample::Create()` function to the `Dali::WidgetApplication::InitSignal()` function:
+   ```
+    class WidgetApplicationExample : public ConnectionTracker
+    {
+      public:
+        WidgetApplicationExample( WidgetApplication& application )
+          : mApplication( application )
+        {
+          mApplication.InitSignal().Connect( this, &WidgetApplicationExample::Create );
+        }
+
+        void Create( Application& application )
+        {
+          // Register widget creating function here.
+        }
+
+      private:
+        WidgetApplication& mApplication;
+    }
+   ```
+
+3. Create own widget class
+
+   On starting, ensure that you must have background knowledge on [Handle/Body Pattern: Basic Way of Using DALi Objects](handle.md).
+
+   Additionally, for this pattern you need to construct two classes for custom widget:
+
+   * Handle class
+   * Body class
+
+   In this step, we will implement `Demo::SampleWidget` class and `Demo::Internal::SampleWidget` class as handle/body class.
+
+   1. Create handle class of custom widget by including <sample-widget.h> header file in your application:
+   ```
+   #include <dali/public-api/adaptor-framework/widget.h>
+
+   namespace Demo
+   {
+   namespace Internal
+   {
+     class SampleWidget;
+   }
+   class SampleWidget : public Dali::Widget    // Inherit widget class
+   {
+   public:
+     SampleWidget();
+     static SampleWidget New();
+     ~SampleWidget();
+     SampleWidget( const SampleWidget& sampleWidget );
+     SampleWidget& operator=( const SampleWidget& sampleWidget );
+   public:
+     explicit SampleWidget( Internal::SampleWidget* sampleWidget );
+   };
+   }
+   ```
+
+   2. Create body class of custom widget by using <sample-widget-impl.h> header file in your application:
```

Comment:
should we use code tag for header file name?

`<sample-widget-impl.h>` header file